### ISSUE TITLE
Sia v1.3.0 mining error.

### ISSUE THREAD

**Issue by miningpoolhub** [NebulousLabs/Sia#2277]
I'm running miningpoolhub.

One of my server is running 1.0.4 LTS version of siad and one is running 1.3.0 the latest release.
I had no problem with 1.0.4 LTS version mining, but same pool software doesn't produce block with 1.3.0
What should I have to change with this version?

Here's the log I got from 1.3.0

2017/08/30 09:40:17.811033 blockmanager.go:145: CRITICAL: ERROR: an invalid block was submitted: transaction spends a nonexisting siacoin output
2017/08/30 09:40:17.811218 blockmanager.go:210: ERROR returned by managedSubmitBlock: transaction spends a nonexisting siacoin output
2017/08/30 12:40:04.742093 blockmanager.go:145: CRITICAL: ERROR: an invalid block was submitted: transaction spends a nonexisting siacoin output
2017/08/30 12:40:04.742218 blockmanager.go:210: ERROR returned by managedSubmitBlock: transaction spends a nonexisting siacoin output

I also found that confirmationheight value was changed between those two version. Please don't change API return values easily like this. This breaks many other things.

**Comment by wincss** [comment]
I have same problem. "transaction spends a nonexisting siacoin output"

**Comment by nitronick600** [comment]
@miningpoolhub did you ever figure this out? cc @wincss

**Comment by miningpoolhub** [comment]
@nitronick600 No. I have no idea about it still.

**Comment by DavidVorick** [comment]
@miningpoolhub as I recall, this was due to an error in the transaction pool that we have since fixed in master, but v1.3.1 hasn't been released yet with the update.

**Comment by tbenz9** [comment]
If this is already fixed in master it should probably be closed.  Feel free to reopen if the problem persists after the next release.

Recommendation to devs: close

**Comment by wincss** [comment]
I tried the code from master. same error has occurred.

**Comment by DavidVorick** [comment]
@wincss please join our discord and get on live chat with us to resolve, the hardfork is here and you need to upgrade

**Comment by miningpoolhub** [comment]
Why not release a new version? I checked https://github.com/NebulousLabs/Sia/releases everyday.
I posted this issue on August and no reply for months until the hardfork.
I'm quite surprised now.
Anyway I'll check master branch.

**Comment by DavidVorick** [comment]
We only release when we feel the new versions are ready. We're still working through some issues with the renter, and won't be ready to release until that is complete.

**Comment by miningpoolhub** [comment]
Still not working.
Dev should fix this issue before the hardfork.

**Comment by DavidVorick** [comment]
We need more information, or if possible clear steps to reproduce the issue.

**Comment by miningpoolhub** [comment]
I can confirm that below commit fixes this issue and found blocks successfully.
https://github.com/NebulousLabs/Sia/commit/2a74c55a9adea094f65ab1991fa75333ed54ad43

Thank you for your effort.

**Comment by DavidVorick** [comment]
We have a temporary fix at the branch 'mining-debug'. As a warning, the miner on that branch will start to slow down substantially if the transaction pool starts to grow a lot. We've fixed the bug, and it performs well enough on average, but if Sia has a large spike that code may struggle to keep up.

We will not be merging the 'mining-debug' branch into master, and instead be working on a more complete solution that has acceptable performance. But given that we have limited time before the hardfork triggers, the 'mining-debug' branch will have to do.

We'll try to have the more performant code fixed up by v1.3.1, which was originally intended to come out about a month ago, but got held back.
